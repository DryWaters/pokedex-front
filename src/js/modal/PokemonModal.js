import React from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';

import CloseButton from './CloseButton';

import '../../styles/modal/PokemonModal.css'

const propTypes = {
  modal: PropTypes.bool.isRequired,
  handleCloseModal: PropTypes.func.isRequired,
  id: PropTypes.number,
  name: PropTypes.string,
  description: PropTypes.string,
  species: PropTypes.string,
  forms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      types: PropTypes.arrayOf(PropTypes.string),
      weaknesses: PropTypes.arrayOf(
        PropTypes.shape({
          type: PropTypes.string,
          multiplier: PropTypes.string,
        })
      ),
      ablities: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          hidden: PropTypes.bool,
        })
      ),
      stats: PropTypes.shape({
        hp: PropTypes.number,
        attack: PropTypes.number,
        defense: PropTypes.number,
        'special-attack': PropTypes.number,
        'special-defense': PropTypes.number,
        speed: PropTypes.number,
      }),
      image_path: PropTypes.string,
    })
  ),
  evolutions: PropTypes.shape({
    1: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        types: PropTypes.arrayOf(PropTypes.string),
        image_path: PropTypes.string,
      })
    ),
    2: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        types: PropTypes.arrayOf(PropTypes.string),
        image_path: PropTypes.string,
      })
    ),
    3: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        types: PropTypes.arrayOf(PropTypes.string),
        image_path: PropTypes.string,
      })
    ),
  }),
}

const PokemonModal = (props) => (
  <Modal isOpen={props.modal} className="pokemodal" align="center" external={<CloseButton close={props.handleCloseModal} />}>
    <ModalHeader>Navigate to other Pokemon</ModalHeader>
    <ModalBody>
      <p>Id: {props.id}</p>
      <p>Name: {props.name}</p>
      <p>Description: {props.description}</p>
      <p>Species: {props.species}</p>
    </ModalBody>
  </Modal>
);

PokemonModal.propTypes = propTypes

export default PokemonModal;
