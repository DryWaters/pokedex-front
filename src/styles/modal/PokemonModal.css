.pokemodal {
  width: 1200px;
  max-width: 90%;
  min-width: 400px; /* Needs to fit sprite */
}

.stats-types {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;
}

.flex-item {
  /*max-width: 100%;*/
}
