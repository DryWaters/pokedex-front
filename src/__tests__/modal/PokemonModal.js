import React from 'react';
import { shallow } from 'enzyme';

import PokemonModal from '../../js/modal/PokemonModal';

describe('PokemonModal Component', () => {

  it('should render PokemonModal component without crashing', () => {
    shallow(<PokemonModal handleCloseModal={() => {}} modal={true}/>);
  });

  it('should match PokemonModal component snapshot', () => {
    const wrapper = shallow(<PokemonModal handleCloseModal={() => {}} modal={true}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
});
