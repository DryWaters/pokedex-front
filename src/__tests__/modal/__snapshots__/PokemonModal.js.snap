// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonModal Component should match PokemonModal component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonModal
    handleCloseModal={[Function]}
    modal={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "center",
      "autoFocus": true,
      "backdrop": true,
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": Array [
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          tag="h5"
          wrapTag="div"
>
          Navigate to other Pokemon
</ModalHeader>,
        <ModalBody
          tag="div"
>
          <p>
                    Will need to be edited.
          </p>
          <p>
                    Id: 
          </p>
          <p>
                    Name: 
          </p>
          <p>
                    Description: 
          </p>
          <p>
                    Species: 
          </p>
</ModalBody>,
      ],
      "className": "pokemodal",
      "external": <CloseButton
        close={[Function]}
/>,
      "fade": true,
      "isOpen": true,
      "keyboard": true,
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "charCode": 215,
          "children": "Navigate to other Pokemon",
          "closeAriaLabel": "Close",
          "tag": "h5",
          "wrapTag": "div",
        },
        "ref": null,
        "rendered": "Navigate to other Pokemon",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p>
              Will need to be edited.
</p>,
            <p>
              Id: 
</p>,
            <p>
              Name: 
</p>,
            <p>
              Description: 
</p>,
            <p>
              Species: 
</p>,
            undefined,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Will need to be edited.",
            },
            "ref": null,
            "rendered": "Will need to be edited.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Id: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Id: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Description: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Description: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species: ",
              undefined,
            ],
            "type": "p",
          },
          undefined,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
>
            Navigate to other Pokemon
</ModalHeader>,
          <ModalBody
            tag="div"
>
            <p>
                        Will need to be edited.
            </p>
            <p>
                        Id: 
            </p>
            <p>
                        Name: 
            </p>
            <p>
                        Description: 
            </p>
            <p>
                        Species: 
            </p>
</ModalBody>,
        ],
        "className": "pokemodal",
        "external": <CloseButton
          close={[Function]}
/>,
        "fade": true,
        "isOpen": true,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": "Navigate to other Pokemon",
            "closeAriaLabel": "Close",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": "Navigate to other Pokemon",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Will need to be edited.
</p>,
              <p>
                Id: 
</p>,
              <p>
                Name: 
</p>,
              <p>
                Description: 
</p>,
              <p>
                Species: 
</p>,
              undefined,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Will need to be edited.",
              },
              "ref": null,
              "rendered": "Will need to be edited.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Id: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Id: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Description: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Description: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species: ",
                undefined,
              ],
              "type": "p",
            },
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
